microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricAnnotationDiscoveryImpl.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.microprofile.metrics;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Member;
+import java.util.StringJoiner;
+
+import jakarta.enterprise.inject.spi.configurator.AnnotatedConstructorConfigurator;
+import jakarta.enterprise.inject.spi.configurator.AnnotatedMethodConfigurator;
+import jakarta.enterprise.inject.spi.configurator.AnnotatedTypeConfigurator;
+
+/**
+ * Implementation of metrics annotation discovery event.
+ * <p>
+ *     The {@link jakarta.enterprise.inject.spi.configurator.AnnotatedConstructorConfigurator} and
+ *     {@link jakarta.enterprise.inject.spi.configurator.AnnotatedMethodConfigurator} interfaces share no common ancestor, so
+ *     we have two subtypes of discovery, one for each:
+ *     {@link io.helidon.microprofile.metrics.MetricAnnotationDiscoveryImpl.OfConstructor OfConstructor} and
+ *     {@link MetricAnnotationDiscovery.OfMethod ofMethod}.
+ * </p>
+ *
+ */
+abstract class MetricAnnotationDiscoveryImpl implements MetricAnnotationDiscovery {
```

Comment:
This is not an Impl but an abstract class - please either use `AbstractMetricAnnotationDiscovery` or `MetricAnnotationDiscoveryBase`. The current name is misleading